src/Calculator.cpp

```diff
@@ -6,6 +7,161 @@
  */
 ////////////////////////////////////////////////////////////////////////////
 
+#include <algorithm>
+
 #include "Calculator.h"
 
-// Add Function definitions here.
\ No newline at end of file
+// Add Function definitions here.
+Calculator::Calculator() {}
+
+Calculator::Calculator(int a, int b, Operation operation)
+                      : a_(a), b_(b), operation_(operation)
+{
+    calculateResult();
+}
+
+Calculator::Calculator(const std::string &mathExpression) 
+{
+    std::stringstream expStream(mathExpression);
+    int a, b;
+    char operation;
+
+    expStream >> a >> operation >> b;
+    if(expStream)
+    {
+        a_ = a;
+        b_ = b;
+        if (std::find(validOperations_.begin(), validOperations_.end(), operation) != validOperations_.end())
+        {
+            //is casting here safe? Or would something like a switch statement be better?
+            //we would have to update this code every time we added an operation if we did
+            operation_ = (Operation)operation;
```

Comment:
This is ok for now, in the future the best way is to use static casting, there is a cherno video on it